### DISCUSSION TITLE
Installer Issues

### DISCUSSION THREAD

**Discussion by SchrodingersGat** [inventree/InvenTree#7336] [Q&A] [answered]
Hi everyone,

Just a quick update to note that we are aware that there are some issues that users are experiencing with the [package installer](https://docs.inventree.org/en/stable/start/installer/), causing difficulty installing or upgrading to the latest InvenTree version. The dev team is looking into this and we believe that we have a solution which we can push out shortly.

Thanks everyone for your patience as we try to work though these issues. Distributing a software package is a complex task, and while we do put a lot of effort into ensuring the process is as painless as possible, sometimes it doesn't always go as planned.

At this stage it looks like the [docker](https://docs.inventree.org/en/stable/start/docker/) and [bare metal](https://docs.inventree.org/en/stable/start/install/) install procedures are not impacted.

### References

- https://github.com/inventree/InvenTree/discussions/7235
- https://github.com/inventree/InvenTree/issues/7326
- https://github.com/inventree/InvenTree/issues/7323
- https://github.com/inventree/InvenTree/issues/7304
- https://github.com/inventree/InvenTree/issues/7297
- https://github.com/inventree/InvenTree/issues/7246
- https://github.com/inventree/InvenTree/issues/7236
- https://github.com/inventree/InvenTree/issues/7332
- https://github.com/inventree/InvenTree/issues/7345

**Comment by rohitsingal1** [comment]
Hope to get the resolution soon. Yesterday my work had to be completely stopped. If there is any option for a rollback do let me know.

**Comment by matmair** [comment, marked as answer]
A fix for this was released as 0.15.3.

Depending on the installation method and age, the steps to fix the install issues vary:
- normal installs: run `sudo apt update && sudo apt upgrade` - the site should run again
- DigitalOcean installs:
  - run `sudo apt update && sudo apt upgrade`
  - run `sudo rm -rf /opt/inventree/env/ && export SETUP_EXTRA_PIP="psycopg2-binary pgcli inventree-droplet-plugin" && sudo apt install inventree --reinstall && sudo inventree run cli update`
  - your instance should now work again
  - if you have set up a custom DNS name for your instance you will need to configure this with `inventree config:set INVENTREE_SITE_URL=http://inventree.invenhost.com` - the instance will only answer on the IP listed when you connect to your instance via ssh without that config

If this is not working run `inventree cli` and post the results here.
Please close the issues you opened regarding your setup once they are done

**Comment by rohitsingal1** [comment]
Getting the error when running 
`inventree run cli`

`/opt/inventree/vendor/pkgr/processes/cli: 2: .: cannot open env/bin/activate: No such file`

Receiving this error for all inventree run commands

**Comment by SchrodingersGat** [comment]
To all users who are affected by the installer issues: a new release has just been made (https://github.com/inventree/InvenTree/releases/tag/0.15.3). 

You can track the status of the build here - https://packager.io/gh/inventree/InvenTree/builds/1944 - once that is complete, you can attempt the update again as per the instructions by @matmair above.

**Comment by rohitsingal1** [comment]


**Reply by SchrodingersGat** [reply]
@rohitsingal1 please start a new issue for this, and we can discuss there.

**Reply by rohitsingal1** [reply]
Ok.

**Comment by StormTrooper** [comment]
I upgraded my test server from 0.15.1 to 0.15.3 but the version info says 0.15.4?

```
# Version Information:
InvenTree-Version: 0.15.4
Django Version: 4.2.12



Database: mysql
Debug-Mode: False
Deployed using Docker: False
Platform: Linux-5.4.0-182-generic-x86_64-with-glibc2.31
Installer: None

Active plugins: [{'name': 'InvenTreeBarcode', 'slug': 'inventreebarcode', 'version': '2.0.0'}, {'name': 'InvenTreeCoreNotificationsPlugin', 'slug': 'inventreecorenotificationsplugin', 'version': '1.0.0'}, {'name': 'InvenTreeCurrencyExchange', 'slug': 'inventreecurrencyexchange', 'version': '1.0.0'}, {'name': 'InvenTreeLabel', 'slug': 'inventreelabel', 'version': '1.0.0'}, {'name': 'InvenTreeLabelMachine', 'slug': 'inventreelabelmachine', 'version': '1.0.0'}, {'name': 'InvenTreeLabelSheet', 'slug': 'inventreelabelsheet', 'version': '1.0.0'}, {'name': 'DigiKeyPlugin', 'slug': 'digikeyplugin', 'version': '1.0.0'}, {'name': 'LCSCPlugin', 'slug': 'lcscplugin', 'version': '1.0.0'}, {'name': 'MouserPlugin', 'slug': 'mouserplugin', 'version': '1.0.0'}, {'name': 'TMEPlugin', 'slug': 'tmeplugin', 'version': '1.0.0'}]
```

```
greg@inventree-clone:/opt/inventree/data/backup$ sudo dpkg -l | grep inventree
ii  inventree                             0.15.3-1716822639.1f6e5213.focal  amd64        Open Source Inventory Management System
```

Also upgraded another test server from 0.15.2 to 0.15.3  - same issue. Reports 0.15.4

Apart from that no other issues seen. This time DB migration worked where previously it didn't under  https://github.com/inventree/InvenTree/issues/7345

**Reply by matmair** [reply]
Rhis might be a caching issue, I started a fresh rebuild of the stable packages, should fix it

**Reply by StormTrooper** [reply]
Updated and still occurring. Thought it might be cached my side, but the version is wrong the file:

```
greg@inventree-clone:/opt/inventree/src/backend/InvenTree/InvenTree$ cat version.py | grep INVENTREE_SW_VERSION
INVENTREE_SW_VERSION = '0.15.4'
```

**Reply by matmair** [reply]
What is the result of running `inventree cli`?

**Reply by StormTrooper** [reply]
```
inventree@inventree-clone:/home/greg$ inventree run cli


InvenTree - inventree.org
The Open-Source Inventory Management System


Installation paths:
Base        /opt/inventree
Config      /etc/inventree/config.yaml
Media       /opt/inventree/data/media
Static      /opt/inventree/data/static

Versions:
Python      3.9.5
Django      4.2.12
InvenTree   0.15.4
API         196
Node        N/A
Yarn        N/A

Commit hash:None
Commit date:None

```

**Reply by matmair** [reply]
which branch are you deploying on?

**Reply by StormTrooper** [reply]
Agh. I think that's the issue.  My apt points to 

```deb https://dl.packager.io/srv/deb/inventree/InvenTree/0.15.x/ubuntu 20.04 main```

Should be ```deb https://dl.packager.io/srv/deb/inventree/InvenTree/stable/ubuntu 20.04 main``` for the stable. Correct?

**Reply by matmair** [reply]
depending if you want to decide yourself when the updates run 0.15.x is also reasonable

**Reply by matmair** [reply]
I will try to reproduce

**Reply by matmair** [reply]
Can reproduce

**Reply by matmair** [reply]
Ok this https://github.com/inventree/InvenTree/pull/7353 is the issue - it seems like it should not present a technical issue. @SchrodingersGat should we address this?

**Comment by steftau** [comment]
I can confirm that the 502 bad gateway problem also affects docker updates.
I first tried to update to 0.15.2 and had 502 the problem, solved it by rolling back to 0.14.5.
Now I tried 0.15.3 and the problem still persists.
My docker skills are very limited unfortunately so I can only give limited info.

**Reply by SchrodingersGat** [reply]
@steftau the docker setup does not use the installer package at all, so there should not be any overlap between these particular issues.

However maybe there is a separate issue occurring in your case.

Can you provide any error messages or further info?

Further, the default docker setup no longer uses nginx as the reverse proxy - we have used caddy since version 0.14.0. Have you updated your setup to use the caddyfile? Or still running nginx?

**Reply by steftau** [reply]
Hi Oliver,
thanks for the quick reply!
When I update to the latest stable by running the docker-compose.yml everything installs smoothly. When I try to reload the webpage I get the same 502 bad gateway error. I'm pretty sure that it was an nginx message - so there might be a clue?
When I edit the docker-compose.yml to version 0.14.5 and run the following cmds 

- docker-compose pull
- docker-compose run inventree-server invoke update 
- and docker-compse up -d 

I end up back with 0.14.5 and the page loads without an error.
I hope this brings more light to the issue.
If there's anything more I can find out, with some precise instructions I can maybe help.

**Reply by SchrodingersGat** [reply]
With the setup that causes the 502 error can you provide the output of the following:

```
docker compose logs
```

**Reply by steftau** [reply]
docker-compose logs
Attaching to inventree-worker, inventree-proxy, inventree-server, inventree-db
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:15,400 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:16,919 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:18,529 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:20,311 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:22,510 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:25,389 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:29,720 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:37,253 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:37:51,209 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-server    | Loading config file : /home/inventree/data/config.yaml
inventree-server    | Python version 3.11.9 - /usr/local/bin/python
inventree-server    | 2024-05-28 12:38:17,944 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-db        |
inventree-db        | PostgreSQL Database directory appears to contain a database; Skipping initialization
inventree-db        |
inventree-db        | 2024-05-28 12:36:37.463 UTC [1] LOG:  starting PostgreSQL 13.15 (Debian 13.15-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
inventree-db        | 2024-05-28 12:36:37.464 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
inventree-db        | 2024-05-28 12:36:37.464 UTC [1] LOG:  listening on IPv6 address "::", port 5432
inventree-db        | 2024-05-28 12:36:37.472 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
inventree-db        | 2024-05-28 12:36:37.482 UTC [26] LOG:  database system was shut down at 2024-05-28 12:36:33 UTC
inventree-db        | 2024-05-28 12:36:37.494 UTC [1] LOG:  database system is ready to accept connections
inventree-proxy     | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
inventree-proxy     | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
inventree-proxy     | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
inventree-proxy     | 10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
inventree-proxy     | /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
inventree-proxy     | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
inventree-proxy     | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
inventree-proxy     | /docker-entrypoint.sh: Configuration complete; ready for start up
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: using the "epoll" event method
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: nginx/1.26.0
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14)
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: OS: Linux 5.4.0-182-generic
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: start worker processes
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: start worker process 20
inventree-proxy     | 2024/05/28 12:37:14 [notice] 1#1: start worker process 21
inventree-proxy     | 2024/05/28 12:37:23 [error] 21#21: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 192.168.1.18, server: , request: "GET /index/ HTTP/1.1", upstream: "http://172.26.0.3:8000/index/", host: "inv.A.com:8000"
inventree-proxy     | 192.168.1.18 - - [28/May/2024:12:37:23 +0000] "GET /index/ HTTP/1.1" 502 559 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36" "-"
inventree-proxy     | 2024/05/28 12:37:23 [error] 21#21: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 192.168.1.18, server: , request: "GET /favicon.ico HTTP/1.1", upstream: "http://172.26.0.3:8000/favicon.ico", host: "inv.A.com:8000", referrer: "http://inv.A.com:8000/index/"
inventree-proxy     | 192.168.1.18 - - [28/May/2024:12:37:23 +0000] "GET /favicon.ico HTTP/1.1" 502 559 "http://inv.A.com:8000/index/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36" "-"
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:16,608 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:18,420 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:20,183 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:22,104 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:24,333 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:27,352 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:31,931 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:39,704 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:37:53,872 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL
inventree-worker    | Loading config file : /home/inventree/data/config.yaml
inventree-worker    | Python version 3.11.9 - /usr/local/bin/python3
inventree-worker    | 2024-05-28 12:38:20,860 ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL

**Reply by steftau** [reply]
![502](https://github.com/inventree/InvenTree/assets/110892901/46cbdf27-a0b0-4a18-8b4a-111553b499ad)

**Reply by wolflu05** [reply]
Maybe that's still the old Nginx from the package installer, that is responding here?

**Reply by steftau** [reply]
Maybe my docker-compose.yaml is to blame. nginx still seems to be the proxy there:

> # Uses gunicorn as the web server
    inventree-server:
        container_name: inventree-server
        # If you wish to specify a particular InvenTree version, do so here
        image: inventree/inventree:stable
        expose:
            - 8000
        depends_on:
            - inventree-db
        volumes:
            # Data volume must map to /home/inventree/data
            - data:/home/inventree/data
        env_file:
            # Environment variables required for the production server are configured in prod-config.env
            - prod-config.env
        restart: unless-stopped

    # Background worker process handles long-running or periodic tasks
    inventree-worker:
        container_name: inventree-worker
        # If you wish to specify a particular InvenTree version, do so here
        image: inventree/inventree:stable
        command: invoke worker
        depends_on:
            - inventree-db
            - inventree-server
        volumes:
            # Data volume must map to /home/inventree/data
            - data:/home/inventree/data
        env_file:
            # Environment variables required for the production server are configured in prod-config.env
            - prod-config.env
        restart: unless-stopped

    # nginx acts as a reverse proxy
    # static files are served directly by nginx
    # media files are served by nginx, although authentication is redirected to inventree-server
    # web requests are redirected to gunicorn
    # NOTE: You will need to provide a working nginx.conf file!
    inventree-proxy:
        container_name: inventree-proxy
        image: nginx:stable
        depends_on:
            - inventree-server
        ports:
            # Change "1337" to the port that you want InvenTree web server to be available on
            - 192.168.1.133:8000:80
        volumes:
            # Provide ./nginx.conf file to the container
            # Refer to the provided example file as a starting point
            - ./nginx.conf:/etc/nginx/conf.d/default.conf:ro
            # nginx proxy needs access to static and media files
            - data:/var/www
        restart: unless-stopped

volumes:
    # NOTE: Change /path/to/data to a directory on your local machine
    # Persistent data, stored external to the container(s)
    data:
        driver: local
        driver_opts:
            type: none
            o: bind
            # This directory specified where InvenTree data are stored "outside" the docker containers
            # Change this path to a local system path where you want InvenTree data stored
            device: /home/xxxxxx

**Comment by SchrodingersGat** [comment]
Pretty clear what is going on here:

`ERROR No CSRF_TRUSTED_ORIGINS specified. Please provide a list of trusted origins, or specify INVENTREE_SITE_URL`

As per our [v0.14.0 update notes](https://inventree.org/blog/2024/03/02/0.14.0) there was a breaking change to the configuration. You need to specify a value for `INVENTREE_SITE_URL`: https://docs.inventree.org/en/stable/start/config/#server-access

**Reply by steftau** [reply]
Thank you very much for your support! I was able to resolve the issue!
Incredible work you guys are doing with InvenTree - very much appreciated!

**Reply by rohitsingal1** [reply]
Yes thanks a ton for your support. The software is amazing and you guys worked on the weekend for the support. Highly appreciable.

**Comment by matmair** [comment]
Most linked issues seem resolved/waiting for customer feedback. Closing and locking to prevent this from becoming a general installer support discussion.

If you have trouble installing, you can read the [install docs](https://docs.inventree.org/en/stable/start/intro/) and open an issue using the [dedicated install trouble template](https://github.com/inventree/InvenTree/issues/new?template=install.yaml).
